### PR TITLE
[Serve] Fix system logging config with REST API

### PR DESCRIPTION
**Pull request by sihanwang41**
## Why are these changes needed?

Fix rest api system logging config

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by sihanwang41** [comment]
@GeneDer @shrekris-anyscale @zcin please take a look, thx!

File: python/ray/serve/scripts.py [hunk 1]
```diff
@@ -763,6 +764,7 @@ def build_app_config(import_path: str, name: str = None):
             "port": DEFAULT_GRPC_PORT,
             "grpc_servicer_functions": grpc_servicer_functions,
         },
+        "logging_config": {**LoggingConfig().dict()},
```
Comment (GeneDer) [inline python/ray/serve/scripts.py:767]: Non-blocker: isn't `LoggingConfig().dict()` already returns a dictionary? Is it necessary to case to unpack to a new dict by doing `{**xyz}`?

File: python/ray/serve/tests/test_config_files/logging_config_test.py [hunk 2]
```diff
@@ -34,6 +36,14 @@ async def __call__(self):
         else:
             log_info["router_log_file"] = None
         log_info["router_log_level"] = logger.level
+
+        try:
+            # Add controller log file path
+            client = _get_global_client()
+            _, log_file_path = ray.get(client._controller._get_logging_config.remote())
+        except Exception:
```
Comment (GeneDer) [inline python/ray/serve/tests/test_config_files/logging_config_test.py:44]: Non-blocker: would be nice to catch the specific exception if we know what it is

**Review by GeneDer — APPROVED** [review]
Some minor comments, but LGTM!

File: python/ray/serve/tests/test_config_files/logging_config_test.py (hunk 2 above)
Comment (shrekris-anyscale) [reply python/ray/serve/tests/test_config_files/logging_config_test.py:44]: +1, we should catch the specific exception here.

File: dashboard/modules/serve/tests/test_serve_dashboard.py [hunk 3]
```diff
@@ -742,5 +743,54 @@ def test_target_capacity_field(ray_start_stop, url: str):
...
+        print(f"*** Starting Iteration {iteration + 1}/{num_iterations} ***\n")
+
+        deploy_config_multi_app(config1, url)
+        wait_for_condition(
+            lambda: requests.get("http://localhost:8000/app").status_code == 200,
+            timeout=15,
+        )
+        print("Deployments are live and reachable over HTTP.\n")
+
+        # Make sure deploymemnt & controller both under json logging
```
Comment (shrekris-anyscale) [inline dashboard/modules/serve/tests/test_serve_dashboard.py:787]:
```suggestion
        # Make sure deploymemnt & controller both log in json format.
```

File: python/ray/serve/scripts.py (hunk 1 above)
Comment (sihanwang41) [reply python/ray/serve/scripts.py]: done! good catch.

File: dashboard/modules/serve/tests/test_serve_dashboard.py [hunk 4]
```diff
@@ -742,5 +743,54 @@ def test_target_capacity_field(ray_start_stop, url: str):
...
+            {
+                "name": "app",
+                "route_prefix": "/app",
+                "import_path": import_path,
+            },
+        ],
+    }
+    # Ensure the REST API is idempotent
+    num_iterations = 2
+    for iteration in range(num_iterations):
```
Comment (shrekris-anyscale) [inline dashboard/modules/serve/tests/test_serve_dashboard.py:777]: Why do we need multiple iterations here? This test seems to deploy an identical config each time?

Comment (sihanwang41) [reply dashboard/modules/serve/tests/test_serve_dashboard.py:777]: looks like it is for testing idempotent for deploying same config multiple times. i follow our style in our test case.

Comment (shrekris-anyscale) [reply dashboard/modules/serve/tests/test_serve_dashboard.py:777]: The other tests check idempotence by applying a new config or deleting a config in each iteration. I don't think we're testing anything important by reapplying the same config.

Comment (sihanwang41) [reply dashboard/modules/serve/tests/test_serve_dashboard.py:777]: got it, removed.

**Review by shrekris-anyscale — APPROVED** [review]
Nice work!

**Review by zcin — APPROVED** [review]

**Pull request merged**